Makefile

```diff
@@ -1,33 +1,35 @@
 ########################
 # Set the program name #
 ########################
-BENCHMARK = propagate
+#BENCHMARK = propagate
 
-#########################################
-# Set macros used for the input program #
-#########################################
-# COMPILER options: pgi, gcc, openarc   #
-# SRCTYPE options: cpp, c               #
-# MODE options: acc, omp, seq           #
-#########################################
-COMPILER ?= pgi
-SRCTYPE ?= cpp
-MODE ?= acc
+###############################################
+#    Set macros used for the input program    #
+###############################################
+# COMPILER options: pgi, gcc, openarc, nvcc   #
+# SRCTYPE options: cpp, c ,cu                 #
+# MODE options: acc, omp, seq, cuda           #
+###############################################
+COMPILER ?= nvcc
+SRCTYPE ?= cu
+MODE ?= cuda
 
 ######################################
 # Set the input source files (CSRCS) #
 ######################################
 ifeq ($(SRCTYPE),cpp)
 ifeq ($(MODE),acc)
-CSRCS = propagate-toz-test_OpenACC.cpp
+CSRCS = propagate-toz-test_OpenACC_v5.cpp
 else
```

Comment:
why do you change version here? and also at lines 24, 28, 32

Reply:
ok. I've taken out the version numbers. However, I was under the impression that those were the lastest, fastest versions. Is that not the case? Those were the versions I had made comparisons to.

Reply:
If that is the case than the changes were fine... but I do not know for sure so I prefer to err on the conservative side. I am going to merge the PR, and we can update later. Thanks and sorry for being picky!